[Zihan] iP

OvOtter opened the pull request:
# Bun 
```"Hello World"```
Bun frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

P.S. Bun is a [bunny](https://knowyourmeme.com/memes/bnuuy) 🐇 who expires to become a programmer, below is his favourite fode:
```c++
unsigned int luck = 0;
while (true) {
  luck += 1;
}
```
> "Get your work done." – Bun

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

OvOtter committed:
Change name to Bun

OvOtter committed:
Complete Level-2

OvOtter committed:
Complete Level-2

OvOtter committed:
Complete Level-3

OvOtter committed:
Complete Level-4

OvOtter committed:
commit class files

OvOtter committed:
testing done

OvOtter committed:
fixed bugs during testing

OvOtter committed:
Level-5 with exceptions (tested)

OvOtter committed:
Level-6

OvOtter committed:
Made enum for commands

OvOtter committed:
Add directory and file to save tasksk

OvOtter committed:
Implement saving tasks to hardist

OvOtter committed:
Edit .gitignore

OvOtter committed:
Implement load tasks feature

OvOtter committed:
Store dates as java.time.LocalDate

OvOtter committed:
Organize code in OOP style

OvOtter committed:
Fix minor bugs

OvOtter committed:
Organize files into package

OvOtter committed:
Merge remote-tracking branch 'origin/add-gradle-support'

adoubleb commented on src/main/java/bun/ui/TaskList.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    public String toStoredContent() {
+        StringBuilder sb = new StringBuilder();
```
Perhaps, it would be more helpful to readers if the variable name meant something concrete?

adoubleb commented on src/main/java/bun/ui/DeleteCommand.java:
```diff
@@ -0,0 +1,16 @@
+package bun.ui;
+
+public class DeleteCommand extends Command {
```
Perhaps, it will be better to use "DeletionCommand"? In order to enforce the noun rule for class names.

adoubleb commented on src/main/java/bun/ui/Parser.java:
```diff
@@ -0,0 +1,75 @@
+            throw new InvalidCommandException(instruction);
+        }
+        switch (command) {
+            case BYE:
```
Shouldn't there be no indentation for switch case?

adoubleb commented on src/main/java/bun/ui/Storage.java:
```diff
@@ -0,0 +1,59 @@
+
+    public ArrayList<Task> load() throws BunException {
+        Path filePath = Path.of(this.filePath);
+        ArrayList<Task> returnedTasks = new ArrayList<>();
```
Good naming for returnedTasks.

adoubleb reviewed (commented):
Overall, the code was easy to understand. Great Job. Few nits. LGTM.

TanXingZhong commented on src/main/java/bun/ui/MissingFieldException.java:
```diff
@@ -0,0 +1,10 @@
+package bun.ui;
+
+public class MissingFieldException extends BunException {
+    protected String varNAme;
```
Perhaps you meant varName ? 🤔

TanXingZhong commented on src/main/java/bun/ui/Parser.java:
```diff
@@ -0,0 +1,75 @@
+        switch (command) {
+            case BYE:
+                return new ExitCommand();
+            case LIST: {
```
Is the indentation here really required? 🤔

TanXingZhong commented on src/main/java/bun/ui/Ui.java:
```diff
@@ -0,0 +1,77 @@
+
+public class Ui {
+    private final static String name = "Bun";
+    private final static String logo = """
```
Nice logo!! 👍

TanXingZhong commented on src/main/java/bun/ui/TaskList.java:
```diff
@@ -0,0 +1,80 @@
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```
I suggest renaming the variable such that it will be easier to understand. 😃

TanXingZhong reviewed (commented):
LGTM!! Just a few minor nits 😄

OvOtter committed:
Create the file for JUnit testing

OvOtter committed:
Fix variable name

OvOtter committed:
Fix indentations

OvOtter committed:
Implement JUnit tests

OvOtter committed:
Fix class name

OvOtter committed:
Add JavaDoc

OvOtter committed:
Reformat code to comply with Java coding standard

OvOtter committed:
Save my local changes

OvOtter committed:
Resolve merge conflicts

OvOtter committed:
Reformat code

OvOtter committed:
Resolve conflicts

OvOtter committed:
Resolve conflicts

OvOtter committed:
Resolve conflicts

OvOtter committed:
Merge branch 'master' into branch-A-JavaDoc

OvOtter committed:
Merge pull request #1 from OvOtter/branch-A-JavaDoc

Branch a java doc

OvOtter committed:
Fix typos

OvOtter committed:
Merge branch 'master' of https://github.com/OvOtter/ip

ChauuuLe reviewed (commented):
There is issue with method too long but overrall the SLAP is good

ChauuuLe commented on src/main/java/bun/ui/Parser.java:
This method handles multiple responsibilities: parsing input, determining command types, and executing logic

ChauuuLe commented on src/main/java/bun/ui/Ui.java:
```diff
@@ -0,0 +1,84 @@
+                     `------'   `-----'    `--'  `--'  \s
+            """;
+
+    public void showWelcome() {
```
Good styling

ChauuuLe commented on src/main/java/bun/ui/Bun.java:
The run() method mixes multiple responsibilities: reading user input, parsing commands, executing logic, and handling errors

ChauuuLe commented on src/main/java/bun/ui/Bun.java:
```diff
@@ -0,0 +1,43 @@
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
```
Consider breaking it into helper methods, such as handleUserInput() or processCommand(), to improve readability and maintainability

OvOtter committed:
Add GUI

OvOtter committed:
Merge branch 'branch-Level-10'

OvOtter committed:
Add assert

Assert statements were added for commands involving index of tasks (must be non-negative).

OvOtter committed:
Reformat code

OvOtter committed:
Merge pull request #2 from OvOtter/branch-A-CodeQuality

Reformat code

OvOtter committed:
Merge pull request #3 from OvOtter/branch-A-Assertion

Add assert

OvOtter committed:
Implement DoAfter task

OvOtter committed:
Merge branch 'master' of https://github.com/OvOtter/ip

OvOtter committed:
Improve parsing of deadline event

OvOtter committed:
Fix codestyle

OvOtter committed:
Fix naming + integrate chatbot into GUI

OvOtter committed:
Update README.md

OvOtter committed:
Overwrite template README.md

OvOtter committed:
Rename UI.png to Ui.png

OvOtter committed:
Update README.md

dawg420 committed:
Restructure source

OvOtter committed:
Revert "Restructure source"

This reverts commit 5fd4e6f3798ca3e43350d7b3ecf9c0e2826e7964.
